src/category_theory/morphism_property.lean

```diff
@@ -172,6 +172,53 @@ end
 
 end naturality_property
 
+section diagonal
+
+variables [has_pullbacks C] {P : morphism_property C}
+
+/-- For `P : morphism_property C`, `P.diagonal` is a morphism property that holds for `f : X ⟶ Y`
+whenever `P` holds for `X ⟶ Y xₓ Y`. -/
+def diagonal (P : morphism_property C) : morphism_property C :=
+λ X Y f, P (pullback.diagonal f)
+
+@[simp]
+lemma diagonal_iff {X Y : C} {f : X ⟶ Y} : P.diagonal f ↔ P (pullback.diagonal f) := iff.rfl
+
+lemma diagonal.respects_iso (hP : P.respects_iso) : P.diagonal.respects_iso :=
```

Comment:
Does it make sense to call this `respects_iso.diagonal`? So that you can write `hP.diagonal`.